garage/tf/models/base.py

```diff
@@ -0,0 +1,313 @@
+"""Base model classes."""
+import abc
+import warnings
+
+import tensorflow as tf
+
+
+class BaseModel(abc.ABC):
```

Comment:
This class needs really comprehensive documentation on the input/output types for each of these methods and properties. do not skimp.

Reply:
i like the docstrings. please document their argument/return types.